carp.client.core/src/commonMain/kotlin/dk/cachet/carp/client/domain/ClientManager.kt

```diff
@@ -28,27 +28,27 @@ abstract class ClientManager<
     /**
      * Determines whether a [DeviceRegistration] has been configured for this client, which is necessary to start adding [StudyRuntime]s.
      */
-    val isConfigured: Boolean get() = repository.deviceRegistration != null
+    suspend fun isConfigured(): Boolean = repository.getDeviceRegistration() != null
 
     /**
      * Configure the [DeviceRegistration] used to register this client device in study deployments managed by the [deploymentService].
      * Use [builder] to configure device-specific registration options, if any.
      */
-    fun configure( builder: TRegistrationBuilder.() -> Unit = {} )
+    suspend fun configure( builder: TRegistrationBuilder.() -> Unit = {} )
     {
         // TODO: Support reconfiguring clients, which will require reregistering the client for all study runtimes.
-        require( !isConfigured ) { "Reconfiguring clients is not supported." }
+        require( !isConfigured() ) { "Reconfiguring clients is not supported." }
 
         val deviceRegistration = createDeviceRegistrationBuilder().apply( builder ).build()
-        repository.deviceRegistration = deviceRegistration
+        repository.setDeviceRegistration( deviceRegistration )
     }
 
     protected abstract fun createDeviceRegistrationBuilder(): TRegistrationBuilder
 
     /**
-     * The studies which run on this client device.
+     * Get the status for the studies which run on this client device.
      */
-    val studies: List<StudyRuntimeStatus> get() = repository.getStudyRuntimeList()
+    suspend fun getStudies(): List<StudyRuntimeStatus> = repository.getStudyRuntimeList()
```

Comment:
It is still possible to cast a StudyRuntimeStatus to a StudyRuntime to access extra information. Do we want to prevent this?

Reply:
Yes. Once the client logic becomes more elaborate and it needs a state machine (like 'deployment' and 'studies' have) I am planning to turn `StudyRuntimeStatus` into a data class.

This is thus temporary.

Reply:
I did see the TODO after I made the comment.